Transportr 2.2.0 crashes after initial setup.

**Describe the bug**
After initial start of the app and selecting any transport network from Germany the app crashes right after granting location permission. It then crashes immediately on every subsequent launching of Transportr. Transportr has been has been fresh installed to verify it's not an update  issue and old data.

**To Reproduce**
Launch the app, select a network and grant location permission. The app will now crash subsequently. 

**Expected behavior**
Normal usage of the app as in the previous version 2.1.5.

**Screenshots**
No screenshots. 

**Versions (please complete the following information):**
 - **Transportr Version:** 2.2.0
 - **Device:** OnePlus 8T
 - **Android Version:** 13.

Crash log:

```
type: crash
osVersion: OnePlus/OnePlus8T/OnePlus8T:13/TQ3A.230901.001/eng.emy.20240107.100827:user/release-keys
package: de.grobox.liberario:124
process: de.grobox.liberario

java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=65537, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {de.grobox.liberario/de.grobox.transportr.map.MapActivity}: java.lang.IllegalArgumentException: provider "network" does not exist
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5339)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5378)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:138)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7924)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:952)
Caused by: java.lang.IllegalArgumentException: provider "network" does not exist
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3015)
	at android.os.Parcel.createException(Parcel.java:2995)
	at android.os.Parcel.readException(Parcel.java:2978)
	at android.os.Parcel.readException(Parcel.java:2920)
	at android.location.ILocationManager$Stub$Proxy.registerLocationListener(ILocationManager.java:1306)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1564)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1247)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1212)
	at de.grobox.transportr.map.PositionController.onActive(PositionController.kt:84)
	at de.grobox.transportr.utils.NotifyingLiveData.onActive(NotifyingLiveData.kt:45)
	at androidx.lifecycle.LiveData.changeActiveCounter(LiveData.java:390)
	at androidx.lifecycle.LiveData$ObserverWrapper.activeStateChanged(LiveData.java:466)
	at androidx.lifecycle.LiveData$LifecycleBoundObserver.onStateChanged(LiveData.java:425)
	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:354)
	at androidx.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.java:196)
	at androidx.lifecycle.LiveData.observe(LiveData.java:205)
	at de.grobox.transportr.map.GpsMapFragment.activateLocationComponent$lambda-6(GpsMapFragment.kt:139)
	at de.grobox.transportr.map.GpsMapFragment.$r8$lambda$bph-Ts8SjpH0ljE8jOsQdWUO9Jw(GpsMapFragment.kt:0)
	at de.grobox.transportr.map.GpsMapFragment$$ExternalSyntheticLambda0.onStyleLoaded(R8$$SyntheticClass:0)
	at com.mapbox.mapboxsdk.maps.MapboxMap.getStyle(MapboxMap.java:120)
	at de.grobox.transportr.map.GpsMapFragment.activateLocationComponent(GpsMapFragment.kt:113)
	at de.grobox.transportr.map.GpsMapFragment.tryActivateLocationComponent(GpsMapFragment.kt:104)
	at de.grobox.transportr.map.GpsMapFragment.onRequestPermissionsResult(GpsMapFragment.kt:177)
	at androidx.fragment.app.FragmentActivity.onRequestPermissionsResult(FragmentActivity.java:769)
	at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:8811)
	at android.app.Activity.dispatchActivityResult(Activity.java:8669)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5332)
	... 13 more
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.internal.util.Preconditions.checkArgument(Preconditions.java:59)
	at com.android.server.location.LocationManagerService.registerLocationListener(LocationManagerService.java:727)
	at android.location.ILocationManager$Stub.onTransact(ILocationManager.java:590)
	at android.os.Binder.execTransactInternal(Binder.java:1280)
	at android.os.Binder.execTransact(Binder.java:1244)


```

--- a/app/src/main/java/de/grobox/transportr/map/PositionController.kt
+++ b/app/src/main/java/de/grobox/transportr/map/PositionController.kt
@@ -78,10 +78,16 @@ import java.util.concurrent.TimeUnit
             PositionState.DISABLED
     }
 
+     private fun isProviderEnabled(provider: String): Boolean {
+         return locationManager.isProviderEnabled(provider)
+     }
+
     @RequiresPermission(ACCESS_FINE_LOCATION)
     override fun onActive() {
         for (provider in LOCATION_PROVIDERS) {
-            locationManager.requestLocationUpdates(provider, MIN_UPDATE_INTERVAL, MIN_UPDATE_DISTANCE, this, Looper.getMainLooper())
+            if (isProviderEnabled(provider)){
+                locationManager.requestLocationUpdates(provider, MIN_UPDATE_INTERVAL, MIN_UPDATE_DISTANCE, this, Looper.getMainLooper())
+            }
         }
     }
 

--- a/app/src/main/java/de/grobox/transportr/map/PositionController.kt
+++ b/app/src/main/java/de/grobox/transportr/map/PositionController.kt
@@ -84,7 +84,7 @@ import java.util.concurrent.TimeUnit
 
     @RequiresPermission(ACCESS_FINE_LOCATION)
     override fun onActive() {
-        for (provider in LOCATION_PROVIDERS) {
+        for (provider in LOCATION_PROVIDERS.filter(locationManager::isProviderEnabled)) {
             if (isProviderEnabled(provider)){
                 locationManager.requestLocationUpdates(provider, MIN_UPDATE_INTERVAL, MIN_UPDATE_DISTANCE, this, Looper.getMainLooper())
             }

--- a/app/src/main/java/de/grobox/transportr/map/PositionController.kt
+++ b/app/src/main/java/de/grobox/transportr/map/PositionController.kt
@@ -78,24 +78,16 @@ import java.util.concurrent.TimeUnit
             PositionState.DISABLED
     }
 
-     private fun isProviderEnabled(provider: String): Boolean {
-         return locationManager.isProviderEnabled(provider)
-     }
-
     @RequiresPermission(ACCESS_FINE_LOCATION)
     override fun onActive() {
         for (provider in LOCATION_PROVIDERS.filter(locationManager::isProviderEnabled)) {
-            if (isProviderEnabled(provider)){
-                locationManager.requestLocationUpdates(provider, MIN_UPDATE_INTERVAL, MIN_UPDATE_DISTANCE, this, Looper.getMainLooper())
-            }
+            locationManager.requestLocationUpdates(provider, MIN_UPDATE_INTERVAL, MIN_UPDATE_DISTANCE, this, Looper.getMainLooper())
         }
     }
 
     @RequiresPermission(ACCESS_FINE_LOCATION)
     override fun onInactive() {
-        for (provider in LOCATION_PROVIDERS) {
-            locationManager.removeUpdates(this)
-        }
+        locationManager.removeUpdates(this)
     }
 
     override fun onLocationChanged(location: Location) {

